sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/MetricsHandler.cs

```diff
@@ -59,16 +74,29 @@ public override void InvokeSync(IExecutionContext executionContext)
         public override async System.Threading.Tasks.Task<T> InvokeAsync<T>(IExecutionContext executionContext)
         {
             executionContext.RequestContext.Metrics.AddProperty(Metric.AsyncCall, true);
+            
+            var operationName = AWSSDKUtils.ExtractOperationName(executionContext.RequestContext.RequestName);
+            var spanName = $"{executionContext.RequestContext.ServiceMetaData.ServiceId}.{operationName}";
+            var span = TracingUtilities.CreateSpanHelper(executionContext.RequestContext, spanName);
+
             try
             {
                 executionContext.RequestContext.Metrics.StartEvent(Metric.ClientExecuteTime);
-                var response = await base.InvokeAsync<T>(executionContext).ConfigureAwait(false);    
-                return response;                  
+                var response = await base.InvokeAsync<T>(executionContext).ConfigureAwait(false);
+
+                span.SetAttribute(TelemetryConstants.RequestIdAttributeKey, executionContext.ResponseContext.Response.ResponseMetadata.RequestId);
+                return response;
+            }
+            catch (Exception ex)
+            {
+                span.SetExceptionAttributes(ex);
+                throw;
             }
             finally
             {
                 executionContext.RequestContext.Metrics.StopEvent(Metric.ClientExecuteTime);
                 this.LogMetrics(executionContext);
+                span.Dispose();
             }            
         }
 
```

Comment:
I intentionally left out adding traces for `AWS_APM_API (.NET Framework 3.5)` because we plan to remove it in v4 and including it would introduce unnecessary complexity to store the spans and retrieve them in the InvokeCallback, which will not be needed in the newer versions.
Additionally, none of our customers will benefit from it as OTel is not supported for .NET Framework versions older than 4.6.2.